rfcs/text/responsiveGrid.md

```diff
@@ -0,0 +1,455 @@
+
+### Advantages
+1. If two [`GridItems`](#GridItem) get the same order, you will still see both.
+1. You can pass a single value for number of units width required.
+1. Order will work as long as numbers are before/after.
+1. Move content with just an adjustment to single GridItem (i.e changes from 6 to 12 units and will auto push next column down)
+
+### Disadvantages
+1. Overlapping modules within a Flexbox layout requires creative use of negative margins, transforms, and/or absolute positioning.
+1. Also has the same accessibility issues around screen readers and keyboard navigation as CSS Grid.
+
+## CSS Framework (Bootstrap or Foundation)
+A CSS framework is a library of predefined CSS styles that a ready to use out-of-the-box. The best known of these are:
+
+[Bootstrap](https://getbootstrap.com/)  
+[Foundation](https://foundation.zurb.com/)
+
+### Advantages
+1. Comprehensive documentation.
+1. Community support.
+1. Potentially quicker development speed.
+
+### Disadvantages
+1. Including a CSS framework in Carbon (a library built with [Styled Components](https://www.styled-components.com/) at its heart) would require generating a special `StyledComponent` at a global level to enable CSS resets and base stylesheets, which may cause conflicts with Carbon's Base Theme. There would be other disadvantages too:
+1. Difficulty in overriding the framework code.
+1. Bloated codebase that attempts to solve compatibility and support issues we don't need.
+1. More complex markup.
+
+# Comparison Methodology
+In order to evaluate the competing CSS styling options (Grid v Flexbox) a simple layout concept was proposed and a solution developed in each approach under consideration.
+## Layout
+![Layout](../images/concept.png)
+
+### CSS Grid
+```html
+<h1>CSS Grid</h1>
+<div class="GridContainer">
+  <div class="GridItem item1" data-start="1" data-end="7" data-row="1">1</div>
+  <div class="GridItem item2" data-start="7" data-end="13" data-row="1">2</div>
+  <div class="GridItem item3" data-start="1" data-end="13" data-row="1">3</div>
+</div>
+
+<hr/>
+
+<div class="GridContainer">
+  <div class="GridItem item1" data-start="1" data-end="13" data-row="1">1</div>
+  <div class="GridItem item2" data-start="1" data-end="13" data-row="2">2</div>
+  <div class="GridItem item3" data-start="1" data-end="13" data-row="3">3</div>
+</div>
+
+<hr/>
+
+<div class="GridContainer">
+  <div class="GridItem item1" data-start="1" data-end="13" data-row="2">1</div>
+  <div class="GridItem item2" data-start="1" data-end="13" data-row="3">2</div>
+  <div class="GridItem item3" data-start="1" data-end="13" data-row="1">3</div>
+</div>
+```
+#### CSS (abridged)
+```css
+.GridContainer {
+  display: grid;
+  grid-gap: 24px;
+  grid-template-columns: repeat(12, 1fr);
+  width: 100%;
+}
+
+.GridItem {
+  box-sizing: border-box;
```

Comment:
Just to check, is this value not passed to all elements by default or has to be added per item?

Reply:
It's by default, this is just for clarity.